pkg/apis/azure/types_infrastructure.go

```diff
@@ -33,6 +33,14 @@ type InfrastructureConfig struct {
 	Zoned bool
 }
 
+// AzResourceReference contains information to identify a generic Azure resource.
+type AzResourceReference struct {
```

Comment:
Why not

```suggestion
type AzureResourceReference struct {
```

? :)

Reply:
golint wrote:
> type name will be used as azure.AzureResourceReference by other packages, and that stutters;
 
:)

But you are right it should be `AzureResourceReference`. I have changed it.

Reply:
Then call it only `ResourceReference`? :)